index.php

```diff
@@ -38,14 +38,24 @@
 $languages = array();
 if ($handle = opendir('lang')) {
     while (false !== ($entry = readdir($handle))) {
-        if ($entry != "." && $entry != "..") {
-             array_push($languages, str_replace(".inc.php", "", $entry));
+        if ($entry != "." && $entry != ".." ) {
+            $entry_lang = str_replace(".inc.php", "", $entry);
+            # Only add language to possibilities if it is the default language or part of the allowed languages 
+            # empty $allowed_lang <=> all languages are allowed
+            if ($entry_lang == $lang || empty($allowed_lang) || in_array($entry_lang, $allowed_lang) ) {
+                array_push($languages, $entry_lang);
+            }
         }
     }
     closedir($handle);
 }
 $lang = detectLanguage($lang, $languages);
-require_once("lang/$lang.inc.php");
+if (file_exists("lang/$lang.inc.php")) {
```

Comment:
> This would leave a blank page, without any message at all for the site admin to investigate what's happening.
Actually it depends on the php configuration. I just realized that the error reporting is configured after and not before, it should be fixed before the language section. I expected the require_once error to be silent in production and displayed and logged when $debug = true.

I don't believe SSP should try to fall back to english when it knows that a configuration/lang file is incorrect or incomplete. But that's my point of view. What do you think @coudot ?

I have a new suggestion, instead of checking if the file exists, before detectLanguage, why not check if $lang is in $languages. If it is not, the language file does not exists so SSP can either : 1) die early and tell there is a configuration error that should be fixed 2) do what you prefer, fall back to english and log

The problem with the current code is that in some edge use cases ex.
Administrator is using english. Most users are japanese. Default language is set to jap instead of ja.
The administrator will notice that the default language is not working on later in production logs when the first non english/japanese use SSP.